hyper/http11/connection.py

```diff
@@ -130,6 +133,15 @@ def request(self, method, url, body=None, headers={}):
         if self._sock is None:
             self.connect()
 
+        #add HTTP Upgrade headers
+        headers[b'connection'] = b'Upgrade, HTTP2-Settings'
+        headers[b'upgrade'] = H2C_PROTOCOL
+        
+        #need to encode SETTINGS frame payload in Base64 and put into the HTTP-2 Settings header 
+        http2_settings = SettingsFrame(0)
+        http2_settings.settings[SettingsFrame.INITIAL_WINDOW_SIZE] = 65535
+        headers[b'HTTP2-Settings'] = base64.b64encode(http2_settings.serialize_body())
```

Comment:
At some point we should add some heuristics about how many times we send these headers. Obviously, at a certain point we know that the upgrade is not going to happen, and we can stop adding the headers. It may be enough to only add them on the first request.

While we're here, let's try to factor this out into a separate function that handles this logic.